src/scene/procedural.js

```diff
@@ -1063,7 +1063,15 @@ function createBox(device, opts) {
 // returns Primitive data, used by ModelComponent and RenderComponent
 function getShapePrimitive(device, type) {
 
-    var primData = shapePrimitiveMap.get(type);
+    // find in cache
+    var primData = null;
+    for (var i = 0; i < shapePrimitives.length; i++) {
+        if (shapePrimitives[i].type === type && shapePrimitives[i].device === device) {
```

Comment:
LOL. So the more apps on the page, the slower each app gets. 😆  But I'm fine going with this to get the deployment out.